gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -82,13 +83,30 @@ public Optional<ProxyBackendConfiguration> getBackendByName(String name)
     @Override
     public void deactivateBackend(String backendName)
     {
-        dao.deactivate(backendName);
+        updateBackendActivationStatus(backendName, () -> dao.deactivate(backendName));
     }
 
     @Override
     public void activateBackend(String backendName)
     {
-        dao.activate(backendName);
+        updateBackendActivationStatus(backendName, () -> dao.activate(backendName));
+    }
+
+    private void updateBackendActivationStatus(String backendName, Runnable action)
+    {
+        GatewayBackend model = dao.findFirstByName(backendName);
+        if (model == null) {
+            throw new IllegalStateException("No backend found with name: " + backendName + ", could not (de)activate");
+        }
+
+        Boolean prevStatus = model.active();
+        action.run();
+        Boolean currStatus = dao.findFirstByName(backendName).active();
```

Comment:
IMO we shouldn't do another lookup here -- remember that this is a database call and thus expensive. We already know the state. Probably we should just accept a boolean `Runnable` and then use that value here?